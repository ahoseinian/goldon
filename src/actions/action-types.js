export const GET_ALL_ITEMS = 'GET_ALL_ITEMS'
export const GET_ALL_ITEMS_FULFILLED = 'GET_ALL_ITEMS_FULFILLED'
export const SET_SEARCH_VALUE = 'SET_SEARCH_VALUE'
export const RESET_SEARCH = 'RESET_SEARCH'
export const ADD_ITEM_TO_CART = 'ADD_ITEM_TO_CART'
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART'
export const TOGGLE_CART_VISIBILITY = 'TOGGLE_CART_VISIBILITY'
export const COMPLETE_CART_ORDER = 'COMPLETE_CART_ORDER'
