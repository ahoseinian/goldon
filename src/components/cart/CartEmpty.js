import React from 'react';
import {Link} from 'react-router'

const CartEmpty = () => (
  <div className="container text-muted text-center p-3">
    <div>محصولی در سبد موجود نمیباشد.</div>
    <Link to="/">بازگشت</Link>
  </div>
);


export default CartEmpty;
