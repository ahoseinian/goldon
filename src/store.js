import {createStore, applyMiddleware} from 'redux'
import goldonApp from './reducers'
// import createLogger from 'redux-logger'
import createPromiseMiddleware from 'redux-promise-middleware'

// const logger = createLogger({collapsed: true});
const promiseMiddleware = createPromiseMiddleware();
const store = createStore(goldonApp, applyMiddleware(promiseMiddleware))


export default store
