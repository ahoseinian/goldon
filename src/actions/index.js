import request from 'superagent'
import {GET_ALL_ITEMS, SET_SEARCH_VALUE, ADD_ITEM_TO_CART, TOGGLE_CART_VISIBILITY, REMOVE_FROM_CART} from './action-types'

export const getAllItems = () => ({
  type: GET_ALL_ITEMS,
  payload: request.get('/api')
})

export const setSearchValue = (field, payload) => ({type: SET_SEARCH_VALUE, field, payload})
export const addItemToCart = (item) => ({type: ADD_ITEM_TO_CART, item})
export const toggleCartVisibility = () => ({type: TOGGLE_CART_VISIBILITY})
export const removeFromCart = (item) => ({type: REMOVE_FROM_CART, item})
