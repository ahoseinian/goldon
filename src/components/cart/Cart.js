import React from 'react'
import {connect} from 'react-redux'
import {browserHistory} from 'react-router'
import {completeCartOrder} from '../../actions'
import CartTable from '../items/TableList'
import getSum from '../../reducers/cart/get-sum'
import Price from '../items/Price'
import CartForm from './CartForm'
import CartEmpty from './CartEmpty'
import SingleRowTable from '../common/SingleRowTable'

const Cart = ({cart, fullPrice, completeCartOrder}) => {
  const handleSubmit = (e) => {
    e.preventDefault();
    const {
      target: {
        elements: {
          tel: {
            value: tel
          },
          address: {
            value: address
          },
          fullname: {
            value: fullname
          }
        }
      }
    } = e;
    const items = cart
      .items
      .map(({id, quantity}) => ({item: id, quantity}))
    completeCartOrder({fullname, tel, address, items}).then(() => browserHistory.push('/cart/finish'))
  }

  return (
    <div>
      <h1 className="text-center py-3 my-3">سبد خرید</h1>
      {cart.items.length
        ? <div className="container">
            <CartTable items={cart.items}/>
            <SingleRowTable name="مجموع">
              <Price value={fullPrice}/>
            </SingleRowTable>
            <CartForm submit={handleSubmit} disabled={!cart.items.length}/>
          </div>
        : <CartEmpty/>}

    </div>
  )
}

export default connect((state) => ({
  cart: state.cart,
  fullPrice: getSum(state.cart.items)
}), {completeCartOrder})(Cart)
