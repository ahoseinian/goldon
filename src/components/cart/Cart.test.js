import React from 'react'
import {shallow, mount} from 'enzyme'
import Cart from './Cart'
import store from '../../store'
import CartEmpty from './CartEmpty'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import {addItemToCart} from '../../actions'
import shortid from 'shortid'

describe('Cart', () => {
  it('renders', () => {
    const wrapper = shallow(<Cart store={store}/>)
    expect(
      console.error.called
    ).toEqual(false)
  })

  it('renders CartEmpty when items are null', () => {
    const wrapper = renderer.create(
        <Cart store={store}/>
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

  it('renders items when it has items', () => {
    store.dispatch(addItemToCart({images:[], price: 2000, id:shortid.generate()}))
    const wrapper = renderer.create(
      <Cart store={store}/>
    ).toJSON()
    expect(wrapper).toMatchSnapshot()
  })

})
