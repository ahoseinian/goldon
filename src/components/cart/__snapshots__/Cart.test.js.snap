exports[`Cart renders CartEmpty when items are null 1`] = `
<div>
  <h1
    className="text-center py-3 my-3">
    سبد خرید
  </h1>
  <div
    className="container text-muted text-center p-3">
    <div>
      محصولی در سبد موجود نمیباشد.
    </div>
    <a
      onClick={[Function]}
      style={Object {}}>
      بازگشت
    </a>
  </div>
</div>
`;

exports[`Cart renders items when it has items 1`] = `
<div>
  <h1
    className="text-center py-3 my-3">
    سبد خرید
  </h1>
  <div
    className="container">
    <div
      className="table-responsive">
      <table
        className="table table-bordered">
        <thead>
          <tr>
            <th>
              <span
                className="fa fa-fw fa-camera" />
            </th>
            <th>
              نام
            </th>
            <th>
              تعداد و قیمت
            </th>
            <th>
              <span
                className="fa fa-fw fa-edit" />
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              className="hzPrvk">
              <img
                className="img-thumbnail"
                role="presentation"
                src="/images/50/undefined.jpg"
                width="60" />
            </td>
            <td
              className="hzPrvk" />
            <td
              className="hzPrvk">
              <div
                className="d-flex">
                <div
                  className="hXDtaZ">
                  1
                </div>
                <div>
                  2,000
                  <small
                    className="hGgYvU">
                    تومان
                  </small>
                </div>
              </div>
              <div>
                2,000
                <small
                  className="hGgYvU">
                  تومان
                </small>
              </div>
            </td>
            <td
              className="text-center hzPrvk"
              onClick={[Function]}>
              <span
                className="fa fa-fw fa-times" />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <table
      className="table table-bordered">
      <tbody>
        <tr>
          <th
            width="120">
            مجموع
          </th>
          <td>
            <div>
              2,000
              <small
                className="hGgYvU">
                تومان
              </small>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
    <form
      onSubmit={[Function]}>
      <div
        className="form-group">
        <label
          htmlFor="fullname">
          نام کامل:
        </label>
        <input
          className="form-control"
          id="fullname"
          name="fullname"
          required={true}
          type="text" />
      </div>
      <div
        className="form-group">
        <label
          htmlFor="tel">
          شماره تلفن:
        </label>
        <input
          className="form-control"
          id="tel"
          name="tel"
          required={true}
          type="number" />
      </div>
      <div
        className="form-group">
        <label
          htmlFor="address">
          آدرس:
        </label>
        <textarea
          className="form-control"
          id="address"
          name="address"
          required={true}
          rows="3" />
      </div>
      <div
        className="text-left">
        <button
          className="btn-lg ejLGtq"
          disabled={false}>
          <span
            className="fa fa-fw fa-check" />
          <span
            className="mx-2">
            ثبت سفارش
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
